File: Copy_String_Remove_Extra_WhiteSpace/Copy.java
/*
    Author : Pakshal Shshikant Jain
    Date : 30/09/2023
    Program : Write a program which accept string from user and copy the
              contents into another string by removing extra white spaces.
                
              Input : "Marvellous        multi          OS"
              Output : "Marvellous multi OS"

*/

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Required Libraries
import java.lang.*;
import java.util.*;

////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Class Declaration
class StringCopy
{
    //Characteristics of Class
    String str;
    char arr[];
    char brr[];

    //Constructor of Class
    StringCopy()
    {
        this.str = " ";
        this.arr = null;
        this.brr = null;
    }

    //Behaviour of Class to Take input from user
    void Accept()
    {
        //Creating object of Scanner Class To Take input from user
        Scanner sobj = new Scanner(System.in);

        //Taking Input from user
        System.out.println("Enter One String");
        this.str = sobj.nextLine();

        this.arr = new char[this.str.length()];
        this.brr = new char[this.str.length()];
    
        this.arr = this.str.toCharArray();
    }

    //Behaviour of Class To Copy String After Removing White Spaces
    void StrCpyX()
    {
        int i = 0;
        int j = 0;

        if((this.arr == null)&&(this.brr == null))
        {
            return;
        }

        //Logic To Copy String According To Given Condition
        while(i != this.arr.length)
        {
            if(this.arr[i] == ' ')
            {
                while((this.arr[i] == ' ')&&(i != this.arr.length))
                {
                    i++;
                }

                this.brr[j] = ' ';
                j++;
            }

            this.brr[j] = this.arr[i];
            j++;
            i++;
        }
    }
}//End of Class

//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Main Class
class Copy
{
    //Main function
    public static void main(String args[])
    {
        System.out.println("Jay Ganesh....");

        //Creating object of StringCopy Class
        StringCopy sobj = new StringCopy();
        //Taking Input from user
        sobj.Accept();
        sobj.StrCpyX();

        //Printing Copied String
        System.out.println("Copied String After Removing White Space is : ");
        System.out.print(sobj.brr);
    }
}

File: Copy_String_Remove_Extra_WhiteSpace/Copy.cpp
/*
    Author : Pakshal Shshikant Jain
    Date : 30/09/2023
    Program : Write a program which accept string from user and copy the
              contents into another string by removing extra white spaces.
                
              Input : Marvellous        multi          OS”
              Output : Marvellous multi OS”

*/

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Required Libraries
#include<iostream>
using namespace std;

///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Class Declaration
class StringCopy
{
    //Characteristics of Class
    private :
        char *src;
        char *dest;

    //Constructor of Class
    public :
        //Parameterised Constructor of Class
        StringCopy(char *SRC,char *DEST)    
        {
            src = SRC;
            dest = DEST;
        }  

    //Behaviour of Class To CopyString After Removing Extra White Spaces
    void StrCpyX()
    {
        if((src == NULL)&&(dest == NULL))
        {
            return;
        }

        //Logic To Remove
        while(*src != '\0')
        {
            if(*src == ' ')
            {
                while((*src == ' ')&&(*src != '\0'))
                {
                    src++;
                }
                *dest = ' ';
                dest++;
            }

            *dest = *src;
            dest++;
            src++;
        }
    }
};//End of Class

///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Main Function
int main()
{
    char arr[50] = {'\0'};
    char brr[50] = {'\0'};

    cout<<"Jay Ganesh...\n";

    //Taking input from user
    cout<<"Enter One String \n";
    scanf("%[^'\n']s",arr);

    //Creating object of StringCopy Class
    StringCopy *sobj = new StringCopy(arr,brr);
    //Call To Behaviours of Class
    sobj->StrCpyX();

    //Printing Copied String
    cout<<"Copied String is : "<<brr<<"\n";

    //Deallocating Memory of Object Created
    delete sobj;

    return 0;
}